packages/miniflare/src/plugins/core/modules.ts

```diff
@@ -156,23 +157,24 @@ function getResolveErrorPrefix(referencingPath: string): string {
 
 export class ModuleLocator {
 	readonly #compiledRules: CompiledModuleRule[];
-	readonly #nodejsCompat: boolean;
+	readonly #nodejsCompat: NodeJSCompatMode;
 	readonly #visitedPaths = new Set<string>();
 	readonly modules: Worker_Module[] = [];
 
 	constructor(
 		private readonly modulesRoot: string,
 		private readonly additionalModuleNames: string[],
 		rules: ModuleRule[] = [],
+		compatibilityDate?: string,
 		compatibilityFlags?: string[]
 	) {
 		// Implicit shallow-copy to avoid mutating argument
 		rules = rules.concat(DEFAULT_MODULE_RULES);
 		this.#compiledRules = compileModuleRules(rules);
-		// `nodejs_compat` doesn't have a default-on date, so we know whether it's
-		// enabled just by looking at flags:
-		// https://github.com/cloudflare/workerd/blob/edcd0300bc7b8f56040d090177db947edd22f91b/src/workerd/io/compatibility-date.capnp#L237-L240
-		this.#nodejsCompat = compatibilityFlags?.includes("nodejs_compat") ?? false;
+		this.#nodejsCompat = getNodeCompatMode(
```

Comment:
nit:
- `#nodejsCompat` is inconsistent with `getNodeCompatMode.mode`
- `getNodeCompatMode().mode` is inconsistent

Maybe `#nodejsCompatMode` and `getNodeCompat` ?

Reply:
An other option being to have 2 distincts `getNodeCompatMode()` and `parseNodeCompatFlags()`

The third option is of course to keep things as they currently are :)

Reply:
I went with the first open. Cheers.